test/unit/test_devices/test_aireos_device.py

```diff
@@ -0,0 +1,522 @@
+import pytest
+from unittest import mock
+
+from pyntc.devices import AIREOSDevice
+from pyntc.devices.aireos_device import (
+    CommandError,
+    OSInstallError,
+    CommandListError,
+    FileTransferError,
+    RebootTimeoutError,
+    NTCFileNotFoundError,
+    convert_filename_to_version,
+)
+
+
+@pytest.mark.parametrize(
+    "filename,version",
+    (("AIR-CT5520-K9-8-8-125-0.aes", "8.8.125.0"), ("AIR-CT5520-8-8-125-0.aes", "8.8.125.0")),
+    ids=("encrypted", "unencrypted"),
+)
+def test_convert_filename_to_version(filename, version):
+    assert convert_filename_to_version(filename) == version
+
+
+def test_enter_config(aireos_device):
+    aireos_device._enter_config()
+    aireos_device.native.config_mode.assert_called()
+
+
+@pytest.mark.parametrize(
+    "filename,expected", (("show_sysinfo.txt", True), ("show_sysinfo_false.txt", False)), ids=("True", "False"),
+)
+def test_image_booted(aireos_show, aireos_boot_image, filename, expected):
+    device = aireos_show([filename])
+    image_booted = device._image_booted(aireos_boot_image)
+    assert image_booted is expected
+
+
+def test_send_command_timing(aireos_send_command_timing):
+    command = "send_command_timing"
+    device = aireos_send_command_timing([f"{command}.txt"])
+    device._send_command(command)
+    device.native.send_command_timing.assert_called()
+    device.native.send_command_timing.assert_called_with(command)
+
+
+def test_send_command_expect(aireos_send_command_expect):
+    command = "send_command_expect"
+    device = aireos_send_command_expect([f"{command}.txt"])
+    device._send_command(command, True, expect_string="Continue?")
+    device.native.send_command_expect.assert_called()
+    device.native.send_command_expect.assert_called_with("send_command_expect", expect_string="Continue?")
+
+
+def test_send_command_error(aireos_send_command_timing):
+    command = "send_command_error"
+    device = aireos_send_command_timing([f"{command}.txt"])
+    with pytest.raises(CommandError):
+        device._send_command(command)
+    device.native.send_command_timing.assert_called()
+
+
+def test_uptime_components(aireos_show):
+    device = aireos_show(["show_sysinfo.txt"])
+    days, hours, minutes = device._uptime_components()
+    assert days == 3
+    assert hours == 2
+    assert minutes == 20
+
+
+@mock.patch.object(AIREOSDevice, "open")
```

Comment:
If we had a test class we would be creating the mock in the setup method. That would be good for a next iteration.